### PR TITLE
PHP8 support

### PR DESCRIPTION
**Pull request by rimvislt**

### CODE REVIEW

**Comment by rimvislt** [comment]
Not sure how to fix coding standards checkout issue

**Comment by alexislefebvre** [comment]
#662 and this issue are duplicates.

File: composer.json [hunk 1]
```diff
@@ -12,15 +12,15 @@
         {"name": "Jonathan H. Wage", "email": "jonwage@gmail.com"}
     ],
     "require": {
-        "php": "^7.2",
-        "doctrine/annotations": "^1.2",
-        "doctrine/mongodb-odm": "^2.0.0",
+        "php": "^7.2 || ^8.0",
```
Comment (dbrumann) [inline composer.json:15]: Would it make sense to use `>=7.2` instead to keep it in line with Symfony's preferred versioning strategy?

File: composer.json [hunk 2]
```diff
@@ -12,15 +12,15 @@
...
+        "doctrine/annotations": "^1.6",
+        "doctrine/mongodb-odm": "^2.2",
```
Comment (dbrumann) [inline composer.json]: Is this really necessary? The old `^2.0.0` would still allow 2.2 for PHP 8, but right now someone with PHP 7.2 would no longer be able to use doctrine/mongodb-odm 2.0/2.1.

File: composer.json (hunk 1 above)
Comment (malarzm) [reply composer.json:15]: No, that's a terrible strategy

File: composer.json (hunk 2 above)
Comment (malarzm) [reply composer.json]: Indeed, the `^2.0.0` constraint should stay (unless there are some blockers but I don't think there'll be any)

Comment (rimvislt) [reply composer.json]: Thank you guys for feedback reverted back to 2.0.0

**Comment by rimvislt** [comment]
Should I update code to minimum php 7.3, because php 7.2 is end of live? (https://www.php.net/supported-versions.php)

**Comment by dbrumann** [comment]
I don't think updating the minimum PHP version should be part of this PR. Since Symfony 5 supports PHP 7.2 it makes sense to keep it unless there is something to be gained from dropping it.

**Comment by malarzm** [comment]
> Should I update code to minimum php 7.3, because php 7.2 is end of live? (https://www.php.net/supported-versions.php)

The best for the bundle is to allow same PHP versions as ODM does: let's keep 7.2 for now. If we are going to drop it we will need to release another ODM version first.

**Comment by malarzm** [comment]
I'm going to close the PR as I've just merged #662. That PR got chosen because it had all checks passing already. Nevertheless I want to thank you @rimvislt for your work on the issue! :)

**Pull request closed without merging**